akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -401,6 +480,15 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
     }
   }
 
+  @tailrec private def writeByteString(bs: ByteString): Unit = {
+    bs match {
+      case cs: ByteString1C ⇒
+        byteBuffer.putInt(bs.length)
+        cs.copyToBuffer(byteBuffer)
+      case _ ⇒ writeByteString(bs.compact)
```

Comment:
I think initially I was doing more stuff here hm... You're right, I'll move back to normal copyToBuffer.

Fun fact: it was super slow until very recently, since it went over the iterator (tons of allocations), but during recent http work I optimised this one 👍